### PR TITLE
Add message formatter class.

### PR DESCRIPTION
**Pull request by kimond**

### CODE REVIEW

File: lib/src/message_formatter.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import 'package:cmustify/src/metadata.dart';
+
+class MessageFormatter {
+  var metaData;
```
Comment (Pacane) [inline lib/src/message_formatter.dart]: add type

File: lib/src/message_formatter.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  setMetaData(Metadata metaData) {
```
Comment (Pacane) [inline lib/src/message_formatter.dart]: No need for this. If you really want to add visibility, make `metadata`and private field (`_metadata`)  and use a setter `void set Metadata(Metadata metadata) { this.metadata = metadata; }`

File: lib/src/message_formatter.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    this.metaData = metaData;
+  }
+
+  String getMessage() {
+    var notificationBody;
```
Comment (Pacane) [inline lib/src/message_formatter.dart]: type

File: lib/src/message_formatter.dart [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    if (this.metaData.getTagValue('title') != null) {
+      notificationBody = this.metaData.getTagValue('title');
+    } else {
+      notificationBody = "Unknown";
+    }
+
+    if (this.metaData.getTagValue('artist') != null) {
+      notificationBody += " by " + this.metaData.getTagValue('artist');
```
Comment (Pacane) [inline lib/src/message_formatter.dart]: use string interpolation

File: lib/src/message_formatter.dart [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+      if (this.metaData.getTagValue('album') != null) {
+        notificationBody += ", " + this.metaData.getTagValue('album');
```
Comment (Pacane) [inline lib/src/message_formatter.dart]: same here

File: test/message_formatter_test.dart [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import 'package:test/test.dart';
+import 'package:cmustify/src/parser.dart';
+import 'package:cmustify/src/message_formatter.dart';
+
+main() {
+  var parser;
```
Comment (Pacane) [inline test/message_formatter_test.dart]: type

File: test/message_formatter_test.dart [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      var result = parser.parse("status playing title Dancing with the stars");
+      final formatter = new MessageFormatter();
+      formatter.setMetaData(result);
+      final message = formatter.getMessage();
+      expect(message, equals("Dancing with the stars"));
+    });
+    test("Should should format data with title and artist", () {
+      var result = parser
+          .parse("status playing title Dancing with the stars artist Jamo");
+      final formatter = new MessageFormatter();
```
Comment (Pacane) [inline test/message_formatter_test.dart]: the formatter could be extracted, and initialized in the setup

File: lib/src/message_formatter.dart (hunk 2 above)
Comment (kimond) [inline lib/src/message_formatter.dart]: So, should I set metadata to public field?

File: test/message_formatter_test.dart [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      final message = formatter.formatMessage(result);
+      expect(message, equals("Dancing with the stars by Jamo"));
+    });
+    test("Should should format data with title, artist and album", () {
+      var result = parser.parse(
+          "status playing title Dancing with the stars artist Jamo album Best hits");
+      final message = formatter.formatMessage(result);
+      expect(message, equals("Dancing with the stars by Jamo, Best hits"));
+    });
+    test("Should should format data with no title, artist and album", () {
```
Comment (Pacane) [inline test/message_formatter_test.dart]: should should

File: lib/src/message_formatter.dart [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import 'package:cmustify/src/metadata.dart';
+
+class MessageFormatter {
+  String formatMessage(Metadata metaData) {
+    var notificationBody;
```
Comment (Pacane) [inline lib/src/message_formatter.dart]: type

File: test/message_formatter_test.dart [hunk 10]
```diff
@@ -12,24 +12,24 @@ main() {
   });
 
   group("Message formatter", () {
-    test("Should should format data with title", () {
+    test("Should format data with title", () {
       var result = parser.parse("status playing title Dancing with the stars");
       final message = formatter.formatMessage(result);
       expect(message, equals("Dancing with the stars"));
```
Comment (Pacane) [inline test/message_formatter_test.dart]: by default the comparator is equals(). So you can just put the string in there.

**Comment by Pacane** [comment]
LGTM

**Pull request merged**